src/scopes.rs

```diff
@@ -1,20 +1,52 @@
-// Copyright 2020 the Deno authors. All rights reserved. MIT license.
-
-use std::cell::RefCell;
-use std::cmp::Eq;
-use std::cmp::PartialEq;
 use std::collections::HashMap;
-use std::fmt::Display;
-use std::hash::Hash;
-use std::hash::Hasher;
-use std::ops::Deref;
-use std::rc::Rc;
-use swc_common::Span;
 use swc_common::DUMMY_SP;
-use swc_ecmascript::ast::ObjectPatProp;
-use swc_ecmascript::ast::Pat;
+use swc_ecmascript::ast::{
+  ArrowExpr, BlockStmt, BlockStmtOrExpr, CatchClause, ClassDecl, DoWhileStmt,
+  FnDecl, ForInStmt, ForOfStmt, ForStmt, Function, Ident,
+  ImportDefaultSpecifier, ImportNamedSpecifier, ImportStarAsSpecifier, Invalid,
+  Module, Param, Pat, SwitchStmt, VarDecl, VarDeclKind, WhileStmt, WithStmt,
+};
+use swc_ecmascript::utils::find_ids;
+use swc_ecmascript::utils::ident::IdentLike;
+use swc_ecmascript::utils::Id;
 use swc_ecmascript::visit::Node;
 use swc_ecmascript::visit::Visit;
+use swc_ecmascript::visit::VisitWith;
+
+#[derive(Debug)]
+pub struct Scope {
+  vars: HashMap<Id, Var>,
+  // TODO(kdy1): Maybe change this to swc_atoms::JsWord
```

Comment:
I removed `swc_atoms` dependency a few days ago, I think we can get away using `String`s

Reply:
I will do so. Is `0.2` ok?